bolt/spark/stack.py

```diff
@@ -79,8 +89,46 @@ def map(self, func):
         func : function 
              This is applied to each value in the intermediate RDD.
         """
-        rdd = self._rdd.map(lambda kv: (kv[0], func(kv[1])))
-        return self._constructor(rdd).__finalize__(self)
+        vshape = self.shape[self.split:]
+        x = random.randn(*vshape)
+        y = random.randn(*vshape)
+        a = asarray([x])
+        b = asarray([x, y])
+        rekey = False
+
+        try:
+            atest = func(a)
+            btest = func(b)
+        except Exception as e:
+            raise RuntimeError("Error evaluating function on test array, got error:\n %s" % e)
```

Comment:
After playing around with this case more I think we should update the core array `map` as well with the same approach, but that can come later.